docs/visualizing-your-sql-queries/plotting-with-seaborn.md

```diff
@@ -1,14 +1,559 @@
+_.extract_asc_to_csv()
+```
+
+If you ran the above cell, you should have a folder `expanded_data` in your current directory that contains the `.csv` files we will be using. In this tutorial, we will be focusing on three of these files: `loan.csv`, `account.csv`, `district.csv`.
+
+## Load Engine
+
+We now load in our SQL extension that allows us to execute SQL queries in Jupyter Notebooks.
+
+```{important}
+<b>Note</b> Ensure you restart any previous notebook that has the same database name as the one initialized below.
+```
+
+```{code-cell} ipython3
+# Loading in SQL extension
+%reload_ext sql
+# Initiating a DuckDB database named 'bank_data.duck.db' to run SQL queries
+%sql duckdb:///bank_data.duck.db
+```
+
+<!-- endregion -->
+
+## Creating Tables
+
+Let's start off with loading three of the eight `.csv` files from the `expanded_data` folder in the current directory to our newly created DuckDB database. Like in the previous tutorial, we will [create a schema](https://ploomber-sql.readthedocs.io/en/latest/intro-to-sql/joining-data-in-sql.html#creating-a-schema) `s1` in which we will store the tables. Here we use the `CREATE TABLE` syntax in DuckDB to ingest four of the eight `.csv` files. The `read_csv_auto` is a function that helps SQL understand our local `.csv` file for creation into our database.
+
+```{code-cell} ipython3
+%%sql
+CREATE SCHEMA s1;
+CREATE TABLE s1.account AS
+FROM read_csv_auto('expanded_data/account.csv', header=True, sep=',');
+CREATE TABLE s1.district AS
+FROM read_csv_auto('expanded_data/district.csv', header=True, sep=',');
+CREATE TABLE s1.loan AS
+FROM read_csv_auto('expanded_data/loan.csv', header=True, sep=',');
+```
+
+The code above will create three tables in the database schema: `s1.account`, `s1.district`, `s1.loan`.
+
+## Exploring the data
+
+Let's take a look at its entries.
+
+```{code-cell} ipython3
+%sqlcmd explore --table s1.account
+```
+
+```{code-cell} ipython3
+%sqlcmd explore --table s1.district
+```
+
+```{code-cell} ipython3
+%sqlcmd explore --table s1.loan
+```
+
+## Matplotlib inheritance
+
+Seaborn is built on top of Matplotlib. Therefore, depending on the seaborn plotting command, it will return either a Matplotlib axes or figure object. If the plotting function is axes-level, a single `matplotlib.pyplot.Axes` object is returned. This object accepts an `ax=` argument, which integrates with Matplotlib's <b>object-oriented interface</b> and allows composing plots with other plots. On the other hand, if the plotting function is figure-level, a `FacetGrid` object is returned. This object, unlike the axes-level object, is more standalone, but "smart" about subplot organization. To learn about these objects in greater detail, visit seaborn's website [here](https://seaborn.pydata.org/tutorial/function_overview.html#figure-level-vs-axes-level-functions).
+
+A few examples denoting this distinction are shown below.
+
+### Axes-level
```

Comment:
Axis (singular), axes (plural)

Please asses whether you are referring to the singular or plural and correct. Run a grammar check

Reply:
"Axes-level" stems from seaborn's documentation and that is what I was referencing throughout the doc. "Axis-level", on the other hand, is not valid because it does not accurately describe the functionality or concept in seaborn.

Reply:
Ok this works